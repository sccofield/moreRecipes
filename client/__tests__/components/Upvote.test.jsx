import React from 'react';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';

import { Upvote, mapDispatchToProps }
  from '../../src/components/Upvote';

describe('Upvote component', () => {
  const props = {
    id: 2,
    upvote: [{
      userId: 2,
      recipeId: 3
    }
    ],
    userId: 2,
    upvoteRecipeActionCreator: jest.fn(() => Promise.resolve()),
  };

  it('should render the component', () => {
    const wrapper = shallow(<Upvote {...props} />);
    expect(wrapper.find('button').exists).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should show that the upvote has been clicked', () => {
    const spy = sinon.spy(Upvote.prototype, 'upvote');
    const wrapper = shallow(<Upvote
      {...props}
    />);
    wrapper.find('#upvoteButton').simulate('click');
    expect(spy.called).toBeTruthy();
  });
});

describe('conatiner functions', () => {
  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch))
      .toHaveProperty('upvoteRecipeActionCreator');

    const { upvoteRecipeActionCreator } = mapDispatchToProps(dispatch);
    upvoteRecipeActionCreator();
    expect(dispatch).toHaveBeenCalled();
  });
});
