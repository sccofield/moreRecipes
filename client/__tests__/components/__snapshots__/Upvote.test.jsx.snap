// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upvote component should render the component 1`] = `
<button
  className="btn btn-outline-info m-1 vote"
  id="upvoteButton"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="fa fa-thumbs-o-up"
  />
   upvote
</button>
`;
